Release 1.0.0 (#2)
kuzzleio/kuzzle-plugin-prometheus: README.md

```

### Configuration

You can find sample configuration files for this plugin and Prometheus scraping job.

#### Plugin

This plugin is configurable using the Kuzzle configuration file `kuzzlerc`.

```json
  "plugins": {
    "kuzzle-plugin-prometheus": {
      "syncInterval": 7500,
      "systemMetricsInterval": 5000
    }
  }
```

The two available options are:
* `syncInterval`: Time interval in __milliseconds__ between two synchronization with Redis.
* `systemMetricsInterval`: Time interval in __milliseconds__ between two system metrics polling.

#### Prometheus

```yaml
global:
  scrape_interval:     5s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.

scrape_configs:
  - job_name: 'kuzzle'
    metrics_path: /_plugin/kuzzle-plugin-prometheus/metrics
    static_configs:
      - targets: ['kuzzleEndpoint:7512'] # 
```

### Demonstration

If you want to simply have a look to a sample Grafana dashboard run the demonstration stack:

```
$ docker-compose -f demo/docker-compose.yml up --scale kuzzle=3
```

This will start a demonstration stack composed with:

> * A three node Kuzzle cluster behind an Nginx loadbalancer.

Reviewer: ```suggestion
* A three node Kuzzle cluster behind an Nginx load balancer.
```

an administration console and a set of plugins that provide advanced functionalities like real-time pub/sub, blazing fast search and geofencing.

* :octocat: __[Github](https://github.com/kuzzleio/kuzzle)__
* :earth_africa: __[Website](https://kuzzle.io)__
* :books: __[Documentation](https://docs.kuzzle.io)__
* :email: __[Gitter](https://gitter.im/kuzzleio/kuzzle)__

### Architecture

Each Kuzzle node expose a route `/metrics` which expose all nodes metrics. To do so, Redis is used to sync metrics data between nodes.
This mechanism make this plugin cluster compatible.

<p align="center">
  <img src="https://user-images.githubusercontent.com/7868838/60268822-979f9580-98ed-11e9-82b4-298edf8d7893.png"/>
</p>

### Installation

To install this plugin on your Kuzzle stack (for each of your Kuzzle nodes):

```
$ git clone https://github.com/kuzzleio/kuzzle-plugin-prometheus.git /path/to/your/kuzzle/plugins/available/kuzzle-plugin-prometheus
$ cd /path/to/your/kuzzle/plugins/available/kuzzle-plugin-prometheus && npm install
$ ln -sr ./ ../../enabled/kuzzle-plugin-prometheus && cd -
```

### Configuration

> You can find sample configuration files for this plugin and Prometheus scraping job in `demo` folder.

Reviewer 2: ```suggestion
You can find sample configuration files for this plugin and the Prometheus scraping job in the `demo` folder.
```

#### Plugin

> This plugin is configurable using the Kuzzle configuration file `kuzzlerc`.

Reviewer 2: ```suggestion
This plugin is configurable using the `kuzzlerc` Kuzzle configuration file.
```

```json
  "plugins": {
    "kuzzle-plugin-prometheus": {
      "syncInterval": 7500,
      "systemMetricsInterval": 5000
    }
  }
```

The two available options are:

> * `syncInterval`: Time interval in __milliseconds__ between two synchronization with Redis.

Reviewer 2: ```suggestion
* `syncInterval`: Time interval in __milliseconds__ between two synchronizations with Redis.
```

* `systemMetricsInterval`: Time interval in __milliseconds__ between two system metrics polling.

#### Prometheus

```yaml
global:
  scrape_interval:     10s # Set the scrape interval to every 10 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.

scrape_configs:
  - job_name: 'kuzzle'
    metrics_path: /_plugin/kuzzle-plugin-prometheus/metrics
    static_configs:
      - targets: ['kuzzleEndpoint:7512'] # 
```

### Demonstration

If you want to simply have a look to a sample Grafana dashboard run the demonstration stack:

```
$ docker-compose -f demo/docker-compose.yml up --scale kuzzle=3
```

This will start a demonstration stack composed with:

> * A three node Kuzzle cluster behind an Nginx load balancer.

Reviewer 2: ```suggestion
* A three nodes Kuzzle cluster behind an Nginx load balancer.
```

* A Prometheus container configured to scrap metrics.
* A Grafana container.

> Once started, go to `http://localhost:3000` and log in with default Grafana credentials:

Reviewer 2: ```suggestion
Once started, go to `http://localhost:3000` and log in with the default Grafana credentials:
```

* username: `admin`
* password: `admin`

> When successfully logged in you can now import demonstration dashboards from `dashboards` folder.

Reviewer 2: ```suggestion
When successfully logged in you can import demonstration dashboards from the `dashboards` folder.
```

> To do so, hover on `+` icon in left sidebar and click on `Import`.Then click on `Upload .json File` and choose the dashboard of your choice. Set up the targeted Prometheus to `Prometheus` and you're done.

Reviewer 2: ```suggestion
To do so, hover on the `+` icon in the left sidebar, and click on `Import`. Then click on `Upload .json File` and choose the dashboard of your choice. Set up the targeted Prometheus to `Prometheus` and you're done.
```

> Make several requests using Kuzzle HTTP API, SDKs or by browsering it with the Admin Console.

Reviewer 2: ```suggestion
Make several requests using Kuzzle's HTTP API or SDKs, or by using the Admin Console.
```
